tutorials/haskell/hspec-golden-test-tutorial/tutorial.md

```diff
@@ -0,0 +1,506 @@
+import           Text.Blaze.Html.Renderer.Pretty        (renderHtml)
+
+
+htmlRendered :: Html -> String
+htmlRendered page = renderHtml page
+
+somePage :: Html
+somePage = html $ do
+    H.head $ do
+        H.title "StackBuilders Tutorial."
+    body $ do
+       "Hello Golden Testers."
+```
+
+
+###### Json.hs
+
+```haskell
+{-# LANGUAGE DeriveGeneric #-}
+
+module JSON.Json where
+
+import            Data.Aeson                   (ToJSON, encode)
+import            GHC.Generics                 (Generic)
+import            Data.ByteString.Lazy         (ByteString)
+
+data Country = Country
+  {
+    cname      :: String,
+    continent :: String,
+    ctag       :: Int
+  } deriving (Generic, Show)
+
+instance ToJSON Country
+
+
+ecuador = Country "Ecuador" "America" 1
+germany = Country "Germany" "Europe" 2
+japan = Country "Japan" "Asia" 3
+
+countries :: [Country]
+countries = [ecuador,germany,japan]
+
+encodeCountries :: [Country] -> ByteString
+encodeCountries = encode
+```
+
+Before coding our tests, lets create and organize `HtmlGoldenSpec.hs` and `JsonGoldenSpec.hs` modules into directories as well.
+
+```shell
+$ tree tests
+test
+├── FizzBuzz
+│   └── FizzBuzzGoldenSpec.hs
+├── Hello
+│   ├── HelloGoldenSpec.hs
+│   └── HelloSpec.hs
+├── Html
+│   └── HtmlGoldenSpec.hs
+├── Json
+│   └── JsonGoldenSpec.hs
+└── Spec.hs
+```
+We are now ready to start coding our golden tests.
+
+#### HTML Golden tests
+
+Lets start with the HTML module importing all the things we need
+
+```haskell
+import           Test.Hspec
+import           Test.Hspec.Golden
+import           HTML.Html
+```
+and writting our test.
+
+```haskell
+spec :: Spec
+spec =
+    describe "renderHtml" $
+      it "Renders an Html5 file " $
+       defaultGolden "html" (htmlRendered somePage)
+```
+
+Just like the previous example, after running the test we will have a new directory `/.golden/html/` that contains our actual and golden file. But what would happen if we would have to change our HTML template ?
+
+In the practice changes may include several lines and tags but to make a simple example, let's edit the body of our HTML to "Goodbye Golden Testers" and run the tests.    
+
+```shell
+test/Html/HtmlGoldenSpec.hs:11:5:
+  1) Html.HtmlGolden.renderHtml Renders an Html5 file
+      expected: "   
+       <body>
+               Hello Golden Testers.
+       </body>"
+      but got: "
+       <body>
+               Goodbye Golden Testers.
+       </body>"
+
+
+Finished in 0.0013 seconds
+3 examples, 1 failure
+```
+
+Our HTML test failed, but not big deal, we can edit our `golden` file and things will work just fine. Pretty simple right?
+
+Actually its not, let{s remeber this is just a didactic example, in the real life golden files stores really large outputs and it would be a waste of time to update them by hand. Fortunately `hspec-golden` provides us with `hgold` CLI tool that automatically updates golden files using. Let's install it.
```

Comment:
@Jagl257 _let's_ (apostrophe) and _remember_ (m before b)

Reply:
@CristhianMotoche yes I reviewed the hole document and chaged the lets to let's. Thanks Cris.